cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -961,6 +966,89 @@ public void deleteFlowQueues(HttpRequest request, HttpResponder responder,
     }
   }
 
+  @GET
+  @Path("/apps/{app-id}/services/{service-id}/routeconfig")
+  public void getRouteConfig(HttpRequest request, HttpResponder responder,
+                             @PathParam("namespace-id") String namespaceId,
+                             @PathParam("app-id") String appId,
+                             @PathParam("service-id") String serviceId) throws Exception {
+    try {
+      ProgramId programId = Ids.namespace(namespaceId).app(appId).service(serviceId);
+      if (!store.programExists(programId)) {
+        responder.sendString(HttpResponseStatus.NOT_FOUND, "Service not found");
+        return;
+      }
+
+      ServiceSpecification spec = (ServiceSpecification) lifecycleService.getProgramSpecification(programId);
+      if (spec == null) {
+        responder.sendStatus(HttpResponseStatus.NOT_FOUND);
+        return;
+      }
+
+      RouteConfig routeConfig = routeStore.fetch(programId);
+      if (routeConfig == null) {
+        responder.sendStatus(HttpResponseStatus.OK);
+        return;
+      }
+      responder.sendJson(HttpResponseStatus.OK, routeConfig.getRoutes());
+    } catch (SecurityException e) {
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
```

Comment:
do we need to catch this exception? Doesn't the `HttpExceptionHandler` handle this appropriately?